### PR TITLE
Change training tasks to xray for Jenkins tests

### PR DESCRIPTION
**Pull request by guoyuhong**
## What do these changes do?
1. Change all the training tests to xray for Jenkins tests by passing `RAY_USE_XRAY=1` to Docker environment.
2. `python/ray/rllib/utils/actors.py` directly calls `fetch` which is not supported in raylet. I changed it according to `ray.get`.
3. There are still test failures in Jenkins with check failure of `/ray/thirdparty/build/arrow/cpp/src/plasma/client.cc:628 Check failed: object_entry != objects_in_use_.end()`. I think this does not related to my change. I cannot repro it in my local env, so we leave it for later change. 

I can share some information that I found: 
   - It has message of `WARN: gym.spaces.Box autodetected dtype as <type 'numpy.float32'>. Please provide explicit dtype.` and `Constructing fcnet [256, 256] <class 'torch.nn.modules.activation.Tanh'>
 `, so I add `dtype` to all `spaces.Box`. However, this does not solve the problem.
   - It only fails in A3C with Pytorch environment.
   - Current success rate is about 80%.
   - Examples:
      - https://amplab.cs.berkeley.edu/jenkins/job/Ray-PRB/7061/console , without my change.
      - https://amplab.cs.berkeley.edu/jenkins/job/Ray-PRB/7169/console , without my change.
      - https://amplab.cs.berkeley.edu/jenkins/job/Ray-PRB/7212/console , with my change.
      - https://amplab.cs.berkeley.edu/jenkins/job/Ray-PRB/7231/console , with my change.

### CODE REVIEW

**Comment by ericl** [comment]
Looks like A3C crashed

**Comment by guoyuhong** [comment]
@ericl I know the route cause. It is the same as https://github.com/ray-project/ray/issues/2494 . It looks like there is also the concurrent issue in raylet. We need to consider how to handler it.

**Comment by guoyuhong** [comment]
@AmplabJenkins , retest this, please.

**Comment by guoyuhong** [comment]
@AmplabJenkins , retest this, please.

**Comment by guoyuhong** [comment]
@AmplabJenkins , retest this, please.

**Comment by guoyuhong** [comment]
@AmplabJenkins , retest this, please.

**Comment by guoyuhong** [comment]
@ericl I shared some information in this PR. I think the crashed are only in A3C with pytorch env. Do you know the reason?

**Comment by guoyuhong** [comment]
@AmplabJenkins , retest this, please.

**Comment by ericl** [comment]
@pcmoritz, is this perhaps related to the pytorch crash on init we saw before?

**Comment by robertnishihara** [comment]
@ericl @guoyuhong From the log I see the error

```
/ray/thirdparty/build/arrow/cpp/src/plasma/client.cc:628 Check failed: object_entry != objects_in_use_.end() 
```

which is probably different from the pytorch crash (which should be fixed now).

File: python/ray/rllib/utils/actors.py [hunk 1]
```diff
@@ -36,7 +36,11 @@ def completed_prefetch(self):
 
         for worker, obj_id in self.completed():
             plasma_id = ray.pyarrow.plasma.ObjectID(obj_id.id())
-            ray.worker.global_worker.plasma_client.fetch([plasma_id])
+            if not ray.global_state.use_raylet:
+                ray.worker.global_worker.plasma_client.fetch([plasma_id])
+            else:
+                ray.worker.global_worker.local_scheduler_client\
```
Comment (robertnishihara) [inline python/ray/rllib/utils/actors.py]: minor, but let's avoid using `\` and do this with parentheses, e.g., something like

```python
(ray.worker.global_worker.local_scheduler_client
    .reconstruct_objects([obj_id], True))
```

or whatever `yapf` insists on. 

From https://www.python.org/dev/peps/pep-0008/

> The preferred way of wrapping long lines is by using Python's implied line continuation inside parentheses, brackets and braces. Long lines can be broken over multiple lines by wrapping expressions in parentheses. These should be used in preference to using a backslash for line continuation.

**Comment by guoyuhong** [comment]
@robertnishihara This check failure may be caused by some undesirable use from script, e.g. https://amplab.cs.berkeley.edu/jenkins/job/Ray-PRB/7206/console . This example showed a failure of `/ray/thirdparty/build/arrow/cpp/src/plasma/client.cc:960 Check failed: manager_conn_ >= 0` which took me a long time to find out the root cause from the calling `fetch` from `python/ray/rllib/utils/actors.py`.

**Comment by guoyuhong** [comment]
@robertnishihara I also have some examples that has this problem which does not have my change:
Examples:
https://amplab.cs.berkeley.edu/jenkins/job/Ray-PRB/7061/console , without my change.
https://amplab.cs.berkeley.edu/jenkins/job/Ray-PRB/7169/console , without my change.
https://amplab.cs.berkeley.edu/jenkins/job/Ray-PRB/7232/console, without my change.

**Comment by guoyuhong** [comment]
@AmplabJenkins , retest this, please.

**Comment by guoyuhong** [comment]
@robertnishihara At the beginning of this PR, I tried to fix this Plasma Client check failure by updating to raylet. However, this does not work and I found a new check failure as described in previous comment. It took me a long time to found that this is a misuse from script. This failure continues after the change. I found some common features that I have met during debugging. And I believe this failure comes from worker instead of backend. This failure only happen in A3C in pytorch, so I start to doubt that this problem is caused by script problem.

File: python/ray/rllib/utils/actors.py [hunk 2]
```diff
@@ -36,7 +36,11 @@ def completed_prefetch(self):
 
         for worker, obj_id in self.completed():
             plasma_id = ray.pyarrow.plasma.ObjectID(obj_id.id())
-            ray.worker.global_worker.plasma_client.fetch([plasma_id])
+            if not ray.global_state.use_raylet:
```
Comment (ericl) [inline python/ray/rllib/utils/actors.py]: Thanks for fixing this!

**Review by robertnishihara — APPROVED** [review]

**Comment by robertnishihara** [comment]
Thanks for this change @guoyuhong! So you're saying that the remaining issues are most likely due to bugs in the actual test scripts?

File: python/ray/rllib/utils/actors.py (hunk 2 above)
Comment (guoyuhong) [reply python/ray/rllib/utils/actors.py]: @ericl Sure. Unfortunately, this won't fix the A3C crash. Maybe you and @pcmoritz have some clue. ;-)

**Pull request merged**